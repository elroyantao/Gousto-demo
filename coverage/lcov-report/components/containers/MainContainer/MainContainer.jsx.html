<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for components/containers/MainContainer/MainContainer.jsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">components/containers/MainContainer</a> MainContainer.jsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">83.33% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>5/6</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">66.67% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>2/3</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">83.33% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>5/6</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { Route, Switch, BrowserRouter as Router } from 'react-router-dom'
&nbsp;
import Header from '../../presentation/Header/Header'
import CategoryContainer from '../CategoryContainer/CategoryContainer'
import Home from '../../presentation/Home/Home'
&nbsp;
import { fetchCategories } from '../../../actions/categoryAction'
import { fetchProducts } from '../../../actions/productAction'
&nbsp;
@connect(<span class="fstat-no" title="function not covered" >(s</span>tate) =&gt; (<span class="cstat-no" title="statement not covered" >{</span>
  categories: state.categories
}), { fetchCategories, fetchProducts })
export default class MainContainer extends Component {
  static propTypes = {
    categories: PropTypes.arrayOf(PropTypes.shape({
      box_limit: PropTypes.number,
      hidden: PropTypes.bool,
      id: PropTypes.string,
      is_default: PropTypes.bool,
      recently_added: PropTypes.bool,
      title: PropTypes.string
    })),
    fetchCategories: PropTypes.func,
    fetchProducts: PropTypes.func
  }
&nbsp;
  static defaultProps = {
    categories: []
  }
&nbsp;
  componentWillMount() {
    const { fetchCategories, fetchProducts } = this.props
    fetchCategories()
    fetchProducts()
  }
  render() {
    const { categories } = this.props
    return (
      &lt;Router&gt;
        &lt;div&gt;
          &lt;Header categories={categories} /&gt;
          &lt;Switch&gt;
            &lt;Route exact path="/" component={Home} /&gt;
            &lt;Route path="/category/:categoryId" component={CategoryContainer} /&gt;
          &lt;/Switch&gt;
        &lt;/div&gt;
      &lt;/Router&gt;
    )
  }
}
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Thu Jul 20 2017 11:00:38 GMT+0100 (BST)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
