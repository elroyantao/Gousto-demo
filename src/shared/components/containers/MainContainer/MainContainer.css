:global body {
  margin: 0;
}

:global p {
  margin: 0;
}
