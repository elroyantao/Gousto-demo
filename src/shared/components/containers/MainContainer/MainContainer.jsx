import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { Route, Switch, BrowserRouter as Router } from 'react-router-dom'

import Header from '../../presentation/Header/Header'
import CategoryContainer from '../CategoryContainer/CategoryContainer'
import Home from '../../presentation/Home/Home'

import { fetchCategories } from '../../../actions/categoryAction'
import { fetchProducts } from '../../../actions/productAction'

@connect((state) => ({
  categories: state.categories
}), { fetchCategories, fetchProducts })
export default class MainContainer extends Component {
  static propTypes = {
    categories: PropTypes.arrayOf(PropTypes.shape({
      box_limit: PropTypes.number,
      hidden: PropTypes.bool,
      id: PropTypes.string,
      is_default: PropTypes.bool,
      recently_added: PropTypes.bool,
      title: PropTypes.string
    })),
    fetchCategories: PropTypes.func,
    fetchProducts: PropTypes.func
  }

  static defaultProps = {
    categories: []
  }

  componentWillMount() {
    const { fetchCategories, fetchProducts } = this.props
    fetchCategories()
    fetchProducts()
  }
  render() {
    const { categories } = this.props
    return (
      <Router>
        <div>
          <Header categories={categories} />
          <Switch>
            <Route exact path="/" component={Home} />
            <Route path="/category/:categoryId" component={CategoryContainer} />
          </Switch>
        </div>
      </Router>
    )
  }
}
