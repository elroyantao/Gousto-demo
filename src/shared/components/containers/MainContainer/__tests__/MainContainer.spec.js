import testComponentHelper from '../../../../lib/test-helper'
import MainContainer from '../MainContainer'

describe('<MainContainer/>', () => {
  const renderComponent = testComponentHelper(MainContainer.WrappedComponent)
  const initialProps = {
    categories: [
      {
        id: '1',
        title: 'one'
      },
      {
        id: '2',
        title: 'two'
      }
    ],
    fetchCategories: jest.fn(),
    fetchProducts: jest.fn()
  }

  describe('@renders', () => {
    it('in default state', () => {
      expect(renderComponent(initialProps).getTree())
        .toMatchSnapshot()
    })
    it('when categories is empty', () => {
      expect(renderComponent({
        ...initialProps,
        categories: []
      }).getTree())
        .toMatchSnapshot()
    })
  })
  describe('@lifecycle', () => {
    describe('componentWillMount', () => {
      let renderedComponent
      beforeAll(() => {
        jest.clearAllMocks()
        renderedComponent = renderComponent(initialProps)
      })
      it('should call fetchProducts', () => {
        const { instance } = renderedComponent
        expect(instance.props.fetchProducts).toHaveBeenCalledTimes(1)
      })
      it('should call fetchCategories', () => {
        const { instance } = renderedComponent
        expect(instance.props.fetchCategories).toHaveBeenCalledTimes(1)
      })
    })
  })
})
