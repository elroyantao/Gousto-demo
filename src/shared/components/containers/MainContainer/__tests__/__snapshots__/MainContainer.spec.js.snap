// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainContainer/> @renders in default state 1`] = `
<BrowserRouter>
  <div>
    <Header
      categories={
        Array [
          Object {
            "id": "1",
            "title": "one",
          },
          Object {
            "id": "2",
            "title": "two",
          },
        ]
      }
    />
    <Switch>
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <Route
        component={[Function]}
        path="/category/:categoryId"
      />
    </Switch>
  </div>
</BrowserRouter>
`;

exports[`<MainContainer/> @renders when categories is empty 1`] = `
<BrowserRouter>
  <div>
    <Header
      categories={Array []}
    />
    <Switch>
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <Route
        component={[Function]}
        path="/category/:categoryId"
      />
    </Switch>
  </div>
</BrowserRouter>
`;
